### ISSUE TITLE
Eslint can't parse rest parameters destructuring.

### ISSUE THREAD

**Issue by PombaM** [eslint/eslint#8625]
**Environment**
* **ESLint Version:** v3.19.0
* **Node Version:** v6.10.3
* **npm Version:** 4.6.1

**What parser (default, Babel-ESLint, etc.) are you using?**
 - default
**Please show your full configuration:**

```
{
	"parserOptions": {
        "ecmaVersion": 6
    },
	"env": {
		"browser": true,
		"es6": true,
		"node": true,
		"mocha": true
	},
	"rules": {
		"array-bracket-spacing": 2,
		"block-spacing": 2,
		"brace-style": [2, "1tbs", {"allowSingleLine": true}],
		"comma-spacing": [2, {"after": true}],
		"curly": 2,
		"dot-notation": [2, {"allowKeywords": true}],
		"eol-last": 2,
		"eqeqeq": 2,
		"guard-for-in": 2,
		"indent": [2, "tab"],
		"keyword-spacing": 2,
		"key-spacing": [2, {"mode": "strict"}],
		"max-len": [2, 150, {"ignoreComments": true}, {"ignoreTrailingComments": true}],
		"object-curly-spacing": 2,
		"object-shorthand": 2,
		"new-parens": 2,
		"no-bitwise": 2,
		"no-caller": 2,
		"no-cond-assign": 2,
		"no-const-assign": 2,
		"no-debugger": 2,
		"no-dupe-args": 2,
		"no-dupe-keys": 2,
		"no-duplicate-case": 2,
		"no-empty": 2,
		"no-eval": 2,
		"no-extend-native": 2,
		"no-extra-semi": 2,
		"no-fallthrough": 2,
		"no-invalid-regexp": 2,
		"no-irregular-whitespace": 2,
		"no-iterator": 2,
		"no-loop-func": 2,
		"no-mixed-spaces-and-tabs": 2,
		"no-multi-spaces": 2,
		"no-multi-str": 2,
		"no-multiple-empty-lines": 2,
		"no-obj-calls": 2,
		"no-process-exit": 2,
		"no-proto": 2,
		"no-redeclare": 2,
		"no-self-assign": 2,
		"no-self-compare": 2,
		"no-script-url": 2,
		"no-spaced-func": 2,
		"no-trailing-spaces": 2,
		"no-undef": 2,
		"no-unexpected-multiline": 2,
		"no-unreachable": 2,
		"no-unused-vars": 2,
		"no-use-before-define": 2,
		"no-useless-concat": 2,
		"no-with": 2,
		"operator-linebreak": [2, "after"],
		"quotes": [2, "single", {"allowTemplateLiterals": true, "avoidEscape": true}],
		"require-await": 2,
		"space-before-blocks": [2, "always"],
		"space-before-function-paren": [2, "never"],
		"space-in-parens": [2, "never"],
		"space-infix-ops": 2,
		"spaced-comment": [2, "always", {"exceptions": ["*"], "markers": ["global"]}],
		"space-unary-ops": [2, {"nonwords": false}],
		"valid-typeof": 2,
		"wrap-iife": [2, "any"]
	}
}

```

**What did you do? Please include the actual source code causing the issue.**

```js
function foo(...[bar = {}, baz]) {
    /* do something */
}
```

**What did you expect to happen?**
Not get any error as it is a correct syntax.

**What actually happened? Please include the actual, raw output from ESLint.**
` Parsing error: Unexpected token [`

**Comment by not-an-aardvark** [comment]
Looks like this is caused by https://github.com/ternjs/acorn/issues/550. It will be fixed when a new release of `acorn` is published.

**Comment by not-an-aardvark** [comment]
Closing because this has been resolved in `acorn`, so it should make it to eslint relatively soon and there's nothing to do on our end.
